### DISCUSSION TITLE
Tag Retention Rule: more complex repository filters

### DISCUSSION THREAD

**Discussion by hostalp** [goharbor/harbor#17846] [Q&A]
I'd like to be able to enter something like the following tag retention rule repository filter:
```
a/b,c/d
```
This doesn't work and gives a HTTP 500 response due to the incorrect synax.
From #14259 and #14510 and also https://github.com/bmatcuk/doublestar#patterns I learned that `a/{b}` should work when dealing with / in the pattern and yes it does really work. However I don't seem to be able to combine both parts of the pattern into one.

- `a/{b},c/{d}` doesn't work (but doesn't cause any syntax error), nothing is found to be processed by such rule
- `{a/{b},c/{d}}` can't be entered into the input field due to changes in #14259 I assume as one of the outer curly brackets is automatically removed
- some other experiments like `{a/{b}},{c/{d}}` were also unsuccessful (can't be entered either)

Is there any way how to enter such expression and get it to work? Or do I really have to create separate rules for each of those variants?

Note that when reading https://github.com/bmatcuk/doublestar#performance I noticed some mention of nested alts so it looks like some form of nesting might work.

**Comment by chlins** [comment]
Why not setup multiple rules to satisfy your requirements?

**Reply by hostalp** [reply]
Because there's a limit for the amount of rules that could be then easily hit (15).
